external/anomaly/ote_anomalib/nncf_task.py

```diff
@@ -0,0 +1,249 @@
+        self,
+        optimization_type: OptimizationType,
+        dataset: DatasetEntity,
+        output_model: ModelEntity,
+        optimization_parameters: Optional[OptimizationParameters] = None,
+    ):
+        """Train the anomaly classification model.
+
+        Args:
+            optimization_type (OptimizationType): Type of optimization.
+            dataset (DatasetEntity): Input dataset.
+            output_model (ModelEntity): Output model to save the model weights.
+            optimization_parameters (OptimizationParameters): Training parameters
+        """
+        logger.info("Optimization the model.")
+
+        if optimization_type is not OptimizationType.NNCF:
+            raise RuntimeError("NNCF is the only supported optimization")
+
+        # config = self.get_config()
+        # logger.info("Training Configs '%s'", config)
+
+        datamodule = OTEAnomalyDataModule(config=self.config, dataset=dataset, task_type=self.task_type)
+        # Setup dataset to initialization of compressed model
+        # datamodule.setup(stage="fit")
+        # nncf_config = yaml.safe_load(OmegaConf.to_yaml(self.config['nncf_config']))
+
+        nncf_callback = NNCFCallback(nncf_config=self.optimization_config["nncf_config"])
+        callbacks = [
+            ProgressCallback(parameters=optimization_parameters),
+            MinMaxNormalizationCallback(),
+            nncf_callback,
+        ]
+
+        self.trainer = Trainer(**self.config.trainer, logger=False, callbacks=callbacks)
+        self.trainer.fit(model=self.model, datamodule=datamodule)
+        self.compression_ctrl = nncf_callback.nncf_ctrl
+        self.save_model(output_model)
+
+        logger.info("Training completed.")
+
+    def save_model(self, output_model: ModelEntity) -> None:
+        """Save the model after training is completed.
+
+        Args:
+            output_model (ModelEntity): Output model onto which the weights are saved.
+        """
+        logger.info("Saving the model weights.")
+        config = self.get_config()
+        model_info = {
+            "compression_state": self.compression_ctrl.get_compression_state(),
+            "meta": {
+                "config": self.config,
+                "nncf_enable_compression": True,
+            },
+            "model": self.model.state_dict(),
+            "config": config,
+            "VERSION": 1,
+        }
+        buffer = io.BytesIO()
+        torch.save(model_info, buffer)
+        output_model.set_data("weights.pth", buffer.getvalue())
+        output_model.set_data("label_schema.json", label_schema_to_bytes(self.task_environment.label_schema))
+        self._set_metadata(output_model)
+
+        f1_score = self.model.image_metrics.F1.compute().item()
+        output_model.performance = Performance(score=ScoreMetric(name="F1 Score", value=f1_score))
+        output_model.precision = self.precision
+
+    def export(self, export_type: ExportType, output_model: ModelEntity) -> None:
```

Comment:
Is this `export` method different than the one implemented in the `InferenceTask`?

Reply:
Yes, another export to onnx
https://github.com/openvinotoolkit/training_extensions/pull/963/files#diff-4a6f509e19303cb2f080de8d73a9a71ff6ffdadd674bf8f5bf95f8c237663c40R245